snapcraft/tests/unit/project_loader/test_config.py

```diff
@@ -2038,6 +2017,20 @@ def test_both_snap_and_prime_specified(self):
             ".*The 'parts/part1' property does not match the required "
             "schema: .* cannot contain both 'snap' and 'prime' keywords.*"))
 
+    def test_missing_required_property_and_missing_adopt_info(self):
+        del self.data['summary']
+        del self.data['adopt-info']
+
+        raised = self.assertRaises(
+            errors.YamlValidationError,
+            project_loader.Validator(self.data).validate)
+
+        expected_message = (
+            "'adopt-info' is a required property or 'summary' is a required "
```

Comment:
now that I see the message here, it feels rather off.

Reply:
Mannnn, this took me hours. Would you prefer what jsonschema gives us?

> Issues while validating snapcraft.yaml: OrderedDict([('name', 'my-snap-name'), ('version', '0.1'), ('description', "This is my-snap's description. You have a paragraph or two to tell the\nmost important story about your snap. Keep it under 100 words though,\nwe live in tweetspace and your description wants to look good in the snap\nstore.\n"), ('grade', 'devel'), ('confinement', 'devmode'), ('parts', OrderedDict([('my-part', OrderedDict([('plugin', 'nil'), ('parse-info', ['foo'])]))]))]) is not valid under any of the given schemas

Just kidding. I'd love to improve it, but couldn't figure out how to do so in a generic manner. Should I hard-code handling for this specific case?